File: 228/main.py
"""
Given a sorted integer array without duplicates, return the summary of its ranges.

For example, given [0,1,2,4,5,7], return ["0->2","4->5","7"].

"""

class Solution(object):
    def summaryRanges(self, nums):
        """
        :type nums: List[int]
        :rtype: List[str]
        """
        res = []
        l = len(nums)
        if l == 0:
            return res
        else:
            nums.append(nums[-1])
            start = nums[0]
            if_range = False
            for i in range(1, l + 1):
                if nums[i] - nums[i - 1] == 1:
                    if_range = True
                else:
                    add = str(start)
                    if if_range:
                        add = add + "->" + str(nums[i - 1])
                    res.append(add)
                    start = nums[i]
                    if_range = False

        return res

ans = Solution()
print(ans.summaryRanges([0,1,2,4,5,7]))

File: 228/main.ts
function summaryRanges(nums: number[]): string[] {
  let result: string[] = [];

  if (nums.length === 0) return [];
  if (nums.length === 1) return [nums[0].toString()];
  let startRange = 0;
  for (let i = 1; i < nums.length; i++) {
    if (nums[i] === nums[i - 1] + 1) {
      continue;
    } else {
      if (i - 1 === startRange) result.push(`${nums[startRange]}`);
      else result.push(`${nums[startRange]}->${nums[i - 1]}`);
      startRange = i;
    }
  }
  if (startRange === nums.length - 1) result.push(`${nums[startRange]}`);
  else result.push(`${nums[startRange]}->${nums[nums.length - 1]}`);
  return result;
}

// loop through the list
// for each number
// if start of Range, continue

console.log(summaryRanges([0, 1, 2, 4, 5, 7]));
console.log(summaryRanges([0, 2, 3, 4, 6, 8, 9]));
